WordPress/gutenberg
# Sidebar: Only render sidebar tabs possessing items to display

Related:
- https://github.com/WordPress/gutenberg/pull/45005

## What?

Reintroduces the original approach in https://github.com/WordPress/gutenberg/pull/45005 which only renders a tab when it contains items. If only one tab has items, those items are rendered directly to the sidebar i.e. no tab list is rendered.

## Why?

This behaviour was removed from the original addition of tabs to increase consistency across blocks. Further explorations have shown this to have limited upside while severely restricting the benefits of tabs. Most blocks will already have consistent tabs (i.e. both settings and appearance tabs). 

While tabs in the block inspector are a Gutenberg experiment, we can lean into them and the original design, seeking widespread concrete feedback, particularly around a11y.

## How?

- Uses the presence of fills for slots belonging to each tab to determine whether the tab should be rendered
- Empty tabs are omitted
- If only one tab has items, they are rendered directly to the sidebar

## Testing Instructions
1. Check that the inspector controls sidebar continues to render correctly while the block inspector tabs experiment is disabled.
2. Turn on the Gutenberg experiment enabling block inspector tabs
3. In the editor, add a variety of blocks and check that each renders the appropriate tabs and controls
4. In particular, add a navigation block and ensure the "list view" tab is rendered
5. Add two paragraph blocks with text. Select both paragraph blocks and confirm the appearance tab's style panels are rendered directly to the block inspector sidebar i.e. there is no tab list.

<details>
<summary>Example block code</summary>

```html



<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. </p>



<p>Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>





<div class="wp-block-cover is-light"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-100 has-background-dim"></span><div class="wp-block-cover__inner-container"></div></div>


<p>Classic block content.</p>
```
</details>

## Screenshots or screencast 


https://user-images.githubusercontent.com/60436221/203515754-dcfc2c63-cdec-4e9b-97e0-efb50ea4e05b.mp4
